pkg/cli/build.go

```diff
@@ -290,6 +287,24 @@ func GenBuildOptions(c *cobra.Command, inputArgs []string, iopts BuildOptions) (
 			iopts.Quiet = true
 		}
 	}
+	var cacheTo *reference.Named
+	var cacheFrom *reference.Named
+	cacheTo = nil
+	cacheFrom = nil
+	if c.Flag("cache-to").Changed {
+		to, err := parse.RepoNameToNamedReference(iopts.CacheTo)
+		if err != nil {
+			return options, nil, nil, err
+		}
+		cacheTo = &to
+	}
+	if c.Flag("cache-from").Changed {
+		from, err := parse.RepoNameToNamedReference(iopts.CacheFrom)
+		if err != nil {
+			return options, nil, nil, err
+		}
+		cacheFrom = &from
+	}
```

Comment:
If the parsing fails, this doesn’t tell the user which option failed.

One possibility would be to pass the option name into `parse.…`; or wrap the error with extra context using `fmt.Errorf(…%w)` here.